ipld/read_test.go

```diff
@@ -365,6 +371,27 @@ func TestGetLeavesByNamespace_MultipleRowsContainingSameNamespaceId(t *testing.T
 
 }
 
+func TestBatchSize(t *testing.T) {
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second)
+	defer cancel()
+
+	bs := blockstore.NewBlockstore(dssync.MutexWrap(ds.NewMapDatastore()))
+	dag := merkledag.NewDAGService(blockservice.New(bs, offline.Exchange(bs)))
+	eds := generateRandEDS(t, 32)
+	_, err := PutData(ctx, ExtractODSShares(eds), dag)
+	require.NoError(t, err)
+
+	out, err := bs.AllKeysChan(ctx)
+	require.NoError(t, err)
+
+	var count int
+	for range out {
+		count++
+	}
+
+	assert.Equal(t, count, batchSize(64))
```

Comment:
Even with the correct batch size via https://github.com/celestiaorg/celestia-node/pull/464/files#r812345568
this won't work. The reason is that `out, err := bs.AllKeysChan(ctx)` will only report different CIDs while here 4096 CIDs seem to always be the same. I'm confused about the latter rn. Could this be due to the way the wrapper handles the parity leaves?